File: C++/38.cpp
class Solution {
public:
    string countAndSay(int n) {
        string res = "1";
        while(n-->1){
            string prev = res;
            res = "";
            auto iter = prev.begin();
            while( iter != prev.end()){
                int count = 1;
                while(*(iter+1) == *iter && iter+1 != prev.end()){
                    iter++;
                    count++;
                }
                res += to_string(count) + *iter;
                iter++;
            }
        }
        return res;
    }
};

File: LC_Python/38.py
class Solution:
    # @param {integer} n
    # @return {string}
    def countAndSay(self, n):
        if not n:
            return ""
        curr = "1"
        for i in range(1,n):
            prev = curr
            curr = ""
            count = 1
            for k in range(1,len(prev)):
                if prev[k] != prev[k-1]:
                    curr += (str(count)+prev[k-1])
                    count = 0
                count += 1
            curr += (str(count)+prev[-1])
        return curr
            